internal/dataplane/parser/ingressrules.go

```diff
@@ -270,9 +271,10 @@ func servicesAllUseTheSameKongAnnotations(
 			valueForThisObject, ok := service.Annotations[k]
 			if !ok {
 				failuresCollector.PushResourceFailure(
-					fmt.Sprintf("in the backend group of %d kubernetes services some have the %s annotation while others don't. "+
-						"this is not supported: when multiple services comprise a backend all kong annotations "+
-						"between them must be set to the same value", len(services), k),
+					fmt.Sprintf("Service has inconsistent %s annotation and is used in multi-Service backend %s. "+
+						"All Services in a multi-Service backend must have matching Kong annotations. Review the "+
+						"associated route resource and align annotations in its multi-Service backends.",
+						k, kongServiceName),
```

Comment:
This is fine as is but we could think about putting stuff like this into either a `func` or the format string into a `const` which could be reused.